service/k8sclient/k8client.go

```diff
@@ -0,0 +1,278 @@
+	var cluster pxc.PerconaXtraDBCluster
+	err := c.kubeCtl.Get(ctx, perconaXtradbClusterKind, params.Name, &cluster)
+	if err != nil {
+		return err
+	}
+
+	cluster.Spec.PXC.Size = params.Size
+	cluster.Spec.ProxySQL.Size = params.Size
+
+	return c.kubeCtl.Apply(ctx, &cluster)
+}
+
+// DeleteXtraDBCluster deletes percona xtradb cluster with provided name.
+func (c *K8Client) DeleteXtraDBCluster(ctx context.Context, params DeleteParams) error {
+	res := &pxc.PerconaXtraDBCluster{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "pxc.percona.com/v1-4-0",
+			Kind:       perconaXtradbClusterKind,
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: params.Name,
+		},
+	}
+	return c.kubeCtl.Delete(ctx, res)
+}
+
+// ListClusters returns list of clusters and their statuses.
+func (c *K8Client) ListClusters(ctx context.Context) ([]Cluster, error) {
+	perconaXtraDBClusters, err := c.getPerconaXtraDBClusters(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: get PSMDB clusters.
+
+	deletingClusters, err := c.getDeletingClusters(ctx, perconaXtraDBClusters)
+	if err != nil {
+		return nil, err
+	}
+	res := append(perconaXtraDBClusters, deletingClusters...)
+
+	return res, nil
+}
+
+// getPerconaXtraDBClusters returns percona xtradb clusters.
+func (c *K8Client) getPerconaXtraDBClusters(ctx context.Context) ([]Cluster, error) {
+	var list meta.List
+	err := c.kubeCtl.Get(ctx, perconaXtradbClusterKind, "", &list)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't get percona xtradb clusters")
+	}
+
+	res := make([]Cluster, len(list.Items))
+	for i, item := range list.Items {
+		var cluster pxc.PerconaXtraDBCluster
+		if err := json.Unmarshal(item.Raw, &cluster); err != nil {
+			return nil, err
+		}
+		val := Cluster{
+			Name:   cluster.Name,
+			Status: string(cluster.Status.Status),
+			Kind:   perconaXtradbClusterKind,
+			Size:   cluster.Spec.ProxySQL.Size,
+		}
+		res[i] = val
+	}
+	return res, nil
+}
+
+// getDeletingClusters returns percona xtradb clusters which are not fully deleted yet.
+func (c *K8Client) getDeletingClusters(ctx context.Context, runningClusters []Cluster) ([]Cluster, error) {
+	var list meta.List
+	err := c.kubeCtl.Get(ctx, "pods", "", &list)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't get kuberneters pods")
+	}
+
+	exists := make(map[string]struct{}, len(runningClusters))
+	for _, cluster := range runningClusters {
+		exists[cluster.Name] = struct{}{}
+	}
+
+	var res []Cluster
+	for _, item := range list.Items {
+		var pod core.Pod
+		if err := json.Unmarshal(item.Raw, &pod); err != nil {
+			return nil, err
+		}
+
+		clusterName := pod.Labels["app.kubernetes.io/instance"]
+		if _, ok := exists[clusterName]; ok {
+			continue
+		}
+
+		var kind string
+		deploymentName := pod.Labels["app.kubernetes.io/name"]
+		switch deploymentName {
+		case "percona-xtradb-cluster":
+			kind = perconaXtradbClusterKind
+		case "psmdb-cluster":
```

Comment:
One more place that should not be there

Reply:
https://github.com/percona-platform/dbaas-controller/pull/21/files#diff-0ebedeb0a1a8df09584e80193b8fcd99R264